### PR TITLE
Improved search

### PR DESCRIPTION
**Pull request by Feichtmeier**
- move the search dialog atop of the left pane
- remove the close button

okokok

@jpnurmi @mivoligo since you both were not 100% happy with the search dialog - what about this instead?

### CODE REVIEW

**Comment by Jupi007** [comment]
This is way much better than the floating popup 👍

**Comment by mivoligo** [comment]
I like this way much more :heart:

**Comment by Feichtmeier** [comment]
light

light theme

**Comment by jpnurmi** [comment]
Getting better. :+1: Have you tried with a transparent barrier color to make it look less modal?

**Comment by jpnurmi** [comment]
Could it be an `Autocomplete` widget? Aren't those commonly used for searching? :slightly_smiling_face:

**Comment by Feichtmeier** [comment]
> Getting better. +1 Have you tried with a transparent barrier color to make it look less modal?

Well I wanted to avoid this design but I know you all want it so here it is :) This is what you guys all prefer, right?

wellwell

I didn't try the autocomplete Widget yet

**Comment by mivoligo** [comment]
Again, we don't have a way to cancel the search using mouse without selecting an option. :)

**Comment by Feichtmeier** [comment]
> 
> 
> Again, we don't have a way to cancel the search using mouse without selecting an option. :)

You can press ESC

**Comment by mivoligo** [comment]
I know but for that I would need to use my keyboard :smiley:

**Comment by Feichtmeier** [comment]
hehe ^^ well when you press the search button you need to use the keyboard because otherwise you can not type in the letters for search :)

File: lib/view/widgets/master_page.dart [hunk 1]
```diff
@@ -30,6 +32,12 @@ class MasterPageState extends State<MasterPage> {
     Future.microtask(() => {goToDetail(menuItems.indexOf(_selectedMenuItem))});
   }
 
+  @override
+  void dispose() {
+    super.dispose();
+    _searchController.dispose();
+  }
```
Comment (mivoligo) [inline lib/view/widgets/master_page.dart:39]: I believe `super.dispose()` should be the last call in dispose function.

**Comment by mivoligo** [comment]
> hehe ^^ well when you press the search button you need to use the keyboard because otherwise you can not type in the letters for search :)

That's fair but what if I just clicked the search button out of curiosity. Maybe I was thinking it's a lens I can grab and move around :wink:

**Comment by Feichtmeier** [comment]
Peek 2021-09-19 19-32

@mivoligo for some reason the barrier dismiss does not work :shrug: 
must be something in that complex dialog hierarchy - since you guys like this design let's get it in and fix the barrier in another PR, ok?

**Comment by mivoligo** [comment]
ok :+1:

File: lib/view/widgets/master_page.dart [hunk 2]
```diff
@@ -48,6 +56,100 @@ class MasterPageState extends State<MasterPage> {
 
     filterItems();
 
+    void openSearchDialog() {
+      showDialog(
+          barrierDismissible: true,
```
Comment (mivoligo) [inline lib/view/widgets/master_page.dart:61]: It's `true` by default so we can omit this.

File: lib/view/widgets/master_page.dart [hunk 3]
```diff
@@ -48,6 +56,100 @@ class MasterPageState extends State<MasterPage> {
...
+          barrierColor: Colors.white.withOpacity(0),
```
Comment (mivoligo) [inline lib/view/widgets/master_page.dart:62]: Maybe better to use `Colors.transparent`

**Comment by mivoligo** [comment]
It looks a bit weird in the narrow view:
Selection_002

**Comment by Feichtmeier** [comment]
Okay this sucks this need to be made better, with less code duplication and with the new navigtion I soon push, closing and re-opening the search issue =)

**Pull request closed without merging**